### PR TITLE
Java source format

### PR DESCRIPTION
**Pull request by tucek**
* changed java formatting to prettier
* updated docs accordingly

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [x] add a line in `doc/news/_preparation_next_release.md`
  - [x] reformat the code with `scripts/dev/reformat-all`
  - [x] make all unit tests pass
  - [x] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [x] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [x] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [x] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [x] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by tucek** [comment]
TODO .restyle.yaml update to current prettier image
TODO consider google-java-format for Java formatting

**Comment by mpranj** [comment]
This PR partially re-does the same work that is already done in #3814 which was done a month ago.
The other PR is just waiting for a release date and is then merged immediately before release.

**Comment by tucek** [comment]
> This PR partially re-does the same work that is already done in #3814 which was done a month ago.
> The other PR is just waiting for a release date and is then merged immediately before release.

Thanks for the heads-up!
What about me basing this PR on your branch?: https://github.com/mpranj/libelektra/tree/clang-format-12 
I would then only have the appropriate changes for updating the java source formatting...

**Comment by mpranj** [comment]
> What about me basing this PR on your branch?

Yes, that sounds great! :heart: 
Then we would not have too many merge conflicts later.

**Comment by tucek** [comment]
@mpranj i think i will migrate to new java formatting after #3869 release, since it will only affect java files anyways

**Comment by markus2330** [comment]
It is better to merge reformatting PRs shortly before releases, otherwise it might create mess in existing PRs. (Releases demand rebases anyway.)

**Comment by tucek** [comment]
This formatting change would only affect java code. If all there are no other PRs than mine, touching the Java code, it would be no problem....

**Comment by markus2330** [comment]
You never know when someone touches somewhere (it might already have happened and the person simply did not create a PR yet). It is really best done shortly before releases. We want to release soon anyway.

**Comment by tucek** [comment]
https://github.com/google/google-java-format is now being used

**Comment by markus2330** [comment]
Can you please separate the commit where you do the changes and where the automatic reformatting is applied?

**Comment by tucek** [comment]
I've reverted the Java source files being formatted with the new formatter. This has to be done just before release of 0.9.8.

**Comment by tucek** [comment]
Ok, added second commit with new Java source format applied and removed task from #4077

File: src/bindings/jna/HelloElektra.java
Comment (mpranj) [inline src/bindings/jna/HelloElektra.java:8]: Is it intentional that there is no space between function name and parenthesis? (as it is different from the rest of Elektra's codebase)

```suggestion
  public static void main (String[] args) {
```

Can we / (should we) add any config, or does this strictly apply the google java style and not allow any config?

File: scripts/dev/reformat-java
Comment (mpranj) [inline scripts/dev/reformat-java:9]: I propose the following changes:

1. Place the jar in `/opt` or some directory that is not writable by everyone.
2. Do not download the dependency in the reformat script, but have it either earlier (e.g. by Dockerfile setup) or not at all (as with the rest of the reformat scripts).

Reasoning:

1. We don't want to introduce security vulnerabilities by executing something anyone could have left in `/tmp`. We also want to persist the dependency somewhere.
2. If we download the dependency in this script, it is executed very late. Temporary network failures may cause the tests to fail late due to download problems. It would be much better and simpler if this is setup somewhere else (Dockerfile, cirrus config, github workflow config...).

Do you want / have time to implement the changes before release? If not, please ping me, then I will do it.

**Review by mpranj — CHANGES_REQUESTED** [review]
Great work! 🚀 
I have a minor, but important change request before we merge this.

File: src/bindings/jna/HelloElektra.java
Comment (tucek) [reply src/bindings/jna/HelloElektra.java:8]: yes, this is intentional and common to most java code formats

Comment (tucek) [reply src/bindings/jna/HelloElektra.java:8]: google java formatter doe not allow any configuration by design

File: scripts/dev/reformat-java
Comment (tucek) [reply scripts/dev/reformat-java:9]: The debian sid dockerfile already includes the jar in `/tmp/google-java-format.jar`
I thought an automated download would help if anyone executes it locally.
I could include the jar (for the debian sid dockerfile) in `/opt/google-java-format.jar` instead of `/tmp/google-java-format.jar`.
Debian sid is the container mentioned to be used for remormatting in `doc/tutorials/run_reformatting_script_with_docker.md`
To prevent the need for manual download, wehn using the script locally, i could introduce a command line switch for the script?
@mpranj what do you think?

Comment (mpranj) [reply scripts/dev/reformat-java:9]: > I could include the jar (for the debian sid dockerfile) in /opt/google-java-format.jar instead of /tmp/google-java-format.jar.

That sounds good!

> To prevent the need for manual download, wehn using the script locally, i could introduce a command line switch for the script?

That's optional to me, but if you want to provide it go ahead 👍

Comment (tucek) [reply scripts/dev/reformat-java]: done

**Review by mpranj — APPROVED** [review]
Thank you so much for the quick fix, LGTM now! 🚀

**Pull request merged**
